flag/file_test.go

```diff
@@ -0,0 +1,94 @@
+package flag
+
+import (
+	"errors"
+	goflag "flag"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/urso/ucfg"
+)
+
+func TestFlagFileParsePrimitives(t *testing.T) {
+	config, err := parseTestFileFlags("-c test.a -c test.b",
+		map[string]FileLoader{
+			".a": makeLoadTestConfig(map[string]interface{}{
+				"b": true,
+				"i": 42,
+			}),
+			".b": makeLoadTestConfig(map[string]interface{}{
+				"b2": true,
+				"f":  3.14,
+				"s":  "string",
+			}),
+		},
+	)
+
+	assert.NoError(t, err)
+	checkFields(t, config)
+}
+
+func TestFlagFileParseOverwrites(t *testing.T) {
+	config, err := parseTestFileFlags("-c test.a -c test.b",
```

Comment:
> So the latest will always overload the previous which I agree is good. What happens if the previous type was different from the next type? I would expect an error.

No error. If types do not match, new value overwrite old value. (e.g. uint -> int). If both old and new value a sub-configs, they are merged recursively ([see mergeConfig](https://github.com/urso/ucfg/blob/master/merge.go#L19)). [Here we overwrite](https://github.com/urso/ucfg/blob/master/merge.go#L32) if old value has primitive type. And [here we overwrite](https://github.com/urso/ucfg/blob/master/merge.go#L38) if old is an object but new value is primitive. When merging/overwriting each config value remembers it's actual source. That is, if unpack fails the error message states the actual file the config has been read from. Introducing some `StrictMerging` config option (or `UnstrictMerging` so strict merging will become default) to check types are compatible when merging sounds like a valid feature request.

>  What happens with array?

Arrays are not appended, but replaced/merged (default behaviour of `(*ucfg.Config).Merge`). ucfg does not really differentiate between array or object (they are represented using same object internally). This simplifies handling arrays and objects a little. Kind of inspired by lua tables. Considering the full paths being generated it kinda makes sense values are overwritten/merged.
e.g. your config 1 is equivalent to `filebeat.prospectors.0.paths.0: a` and config 2 is basically `filebeat.prospectors.0.paths.0: b`. Considering the full path's it makes sense values are overwritten and final config is `filebeat.prospectors.0.paths.0: b`. Merging=overwrite could be potentially relaxed by introducing some `ucfg.AppendArrays` option (despite objects/arrays are similar we can still tell if source was an array) in order to get the later behavior.

Named features requests are not part of flags support itself, but ucfg core functionalty.  They should be added in separate PRs.